app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,31 @@
+class OmniauthCallbacksController < ApplicationController
+  def complete
+    auth = request.env['omniauth.auth']
+    email_address = auth.info.email
+    govuk_one_login_uid = auth.uid
+
+    candidate = Candidate.find_by(govuk_one_login_uid: govuk_one_login_uid) || Candidate.find_by(email_address: email_address)
+
+    if candidate
+      candidate.update!(govuk_one_login_uid: govuk_one_login_uid) if candidate.govuk_one_login_uid.nil?
```

Comment:
This should probably update the email_address as well, with the email from the one login